### PR TITLE
*: change the way of column estimating.

### PR DESCRIPTION
**Pull request by hanfei1991**
@shenli @coocood @zimulala @lamxTyler 
PTAL

### CODE REVIEW

**Comment by shenli** [comment]
@hanfei1991 Please add some description for this PR.

**Comment by hanfei1991** [comment]
When the ddl happens after loading stats. We can estimate as peusdo column for the missed column, instead of making the whole table peusdo.
I havn't processed the case of index. It's the first step.

File: statistics/table.go [hunk 1]
```diff
@@ -166,6 +168,47 @@ func (t *Table) String() string {
 	return strings.Join(strs, "\n")
 }
 
+// ColumnInValid checks if this column is invalid. Exported for test.
```
Comment (lamxTyler) [inline statistics/table.go]: Didn't see it in the test. Is it for future test?

File: statistics/builder.go [hunk 2]
```diff
@@ -134,21 +140,6 @@ func (b *Builder) NewTable() (*Table, error) {
 			}
 		}
 	}
-	// There may be cases that we have no columnSamples, and only after we build the index columns that we can know it is 0,
-	// so we should also checked it here.
-	if t.Count == 0 {
```
Comment (lamxTyler) [inline statistics/builder.go]: Do not need to check here?

File: statistics/table.go (hunk 1 above)
Comment (hanfei1991) [inline statistics/table.go]: Yeah, for future.

File: statistics/builder.go (hunk 2 above)
Comment (hanfei1991) [inline statistics/builder.go]: I have added some tests to cover this.

**Comment by hanfei1991** [comment]
@lamxTyler  add the test for empty table. PTAL

File: statistics/builder.go [hunk 3]
```diff
@@ -134,20 +140,14 @@ func (b *Builder) NewTable() (*Table, error) {
 			}
 		}
 	}
-	// There may be cases that we have no columnSamples, and only after we build the index columns that we can know it is 0,
-	// so we should also checked it here.
 	if t.Count == 0 {
```
Comment (lamxTyler) [inline statistics/builder.go]: Do not need to remove the comment.

File: statistics/builder.go [hunk 4]
```diff
@@ -134,20 +140,14 @@ func (b *Builder) NewTable() (*Table, error) {
...
-				ID:  idx.ID,
-				NDV: pseudoRowCount / 2,
-			}
+		for i := range t.Columns {
+			t.Columns[i] = &Column{ID: b.TblInfo.Columns[i].ID}
+		}
+		for i := range t.Indices {
+			t.Indices[i] = &Column{ID: b.TblInfo.Indices[i].ID}
 		}
+		return t, nil
```
Comment (lamxTyler) [inline statistics/builder.go]: This is redundant.

**Comment by lamxTyler** [comment]
LGTM

File: statistics/table.go [hunk 5]
```diff
@@ -166,6 +168,47 @@ func (t *Table) String() string {
...
+func (t *Table) ColumnInValid(colInfo *model.ColumnInfo) bool {
```
Comment (coocood) [inline statistics/table.go]: `invalid` is a single word.
Rename to `ColumnIsInvalid` is better.

**Pull request merged**